status-handler.js

```diff
@@ -60,6 +62,12 @@ class StatusHandler {
       Status.findOne(ev)
     ]);
 
+    try {
+      await this._dispatchWebhook('build_started', data);
```

Comment:
this needs to not be awaited.  We should not be waiting for some external service(s) to do anything here. it should be fire and forget.  i.e.
```suggestion
      this._dispatchWebhook('build_started', data).catch(err => this.log.error('Status Handler errored %s', err.message, data));
```

Reply:
@msluther i actually think blocking here is good and gives better determinism for the webhooks. it does slow down the processing a bit but we should time it and see if its significant. In general we still have more than 1 concurrency for handling messages for each status-api so we arent really slowing down things that much

Reply:
Blocking
1. Doesn't help determinism because you have multiple handlers reading from the message stream, there's no guarantees about ordering.
1. Makes it so that our service is beholden to whatever performance a webhook happens to have. If that service decides not to respond until they've fully completed some expensive operation, then we're holding onto the message & sockets until that happens. One bad (accidental or otherwise) webhook slows down the whole system and makes it so that other webhooks dont receive their messages in a timely manner. These should all be fire and forget messages.

Reply:
Not sure I understand how `1` is related with the determinism @jcrugzz is talking about. However for `2` each request has a timeout set to `1500` ms. So in the worse case scenario if a web hook does not respond under a certain time, the connection is timeout. Now considering that these requests happen in parallel, I'd say `2` is negligible. Anyway, that said, I do not have strong option on this. I'd say because both @jcrugzz and I are more in favor of the blocking scenario, we should go for it. But again, if @msluther have strong opinion on this, then let's leave it no blocking (since I've already changed it).